Include <png.h> as early as possible to make sure <setjmp.h> is not
included before it.

See http://thread.gmane.org/gmane.linux.distributions.nixos/1501 .

--- inkscape-0.46/src/sp-image.cpp	2008-03-11 05:19:56.000000000 +0100
+++ inkscape-0.46/src/sp-image.cpp	2009-01-26 15:34:33.000000000 +0100
@@ -17,6 +17,8 @@
 # include "config.h"
 #endif
 
+#include <png.h>
+
 #include <cstring>
 #include <string>
 #include <libnr/nr-matrix-fns.h>
@@ -41,7 +43,6 @@
 #include "libnr/nr-matrix-fns.h"
 
 #include "io/sys.h"
-#include <png.h>
 #if ENABLE_LCMS
 #include "color-profile-fns.h"
 #include "color-profile.h"

--- inkscape-0.46/src/dialogs/export.cpp	2008-03-11 05:19:54.000000000 +0100
+++ inkscape-0.46/src/dialogs/export.cpp	2009-01-26 17:01:13.000000000 +0100
@@ -20,6 +20,7 @@
 # include "config.h"
 #endif
 
+#include <png.h>
 #include <gtk/gtk.h>
 #include <gtkmm/box.h>
 #include <gtkmm/buttonbox.h>
@@ -60,7 +61,6 @@
 #include "io/sys.h"
 
 #include "helper/png-write.h"
-#include <png.h>
 
 
 #define SP_EXPORT_MIN_SIZE 1.0

--- inkscape-0.46/src/main.cpp	2008-03-11 05:20:50.000000000 +0100
+++ inkscape-0.46/src/main.cpp	2009-01-26 17:57:12.000000000 +0100
@@ -26,6 +26,7 @@
 #ifdef HAVE_CONFIG_H
 # include "config.h"
 #endif
+#include <png.h>
 #include "path-prefix.h"
 
 #include <gtk/gtkmessagedialog.h>
@@ -111,7 +112,6 @@ using Inkscape::Extension::Internal::Pri
 
 #include "main-cmdlineact.h"
 
-#include <png.h>
 #include <errno.h>
 
 enum {
